### DISCUSSION TITLE
how can I set the tracks select box on the right？

### DISCUSSION THREAD

**Discussion by Jasonzzzzzzzzzzzzzz** [GMOD/jbrowse-components#4001] [General] [answered]
![example](https://github.com/GMOD/jbrowse-components/assets/49664139/848562d2-998a-46df-8096-4d53513ff499)

I just want to set the select box on the right like this exmaple picture. 
I think the code should like this:
"widgets": {
      "hierarchicalTrackSelector": {
        "id": "hierarchicalTrackSelector",
        "type": "HierarchicalTrackSelectorWidget",
        "initialized": true,
        "collapsed": {},
        "view": "4xxiBnXPL"
      }
    },
    "activeWidgets": {
      "hierarchicalTrackSelector": "hierarchicalTrackSelector"
    },

however, it doesn't work.

**Comment by cmdcolin** [comment]
can you clarify what you mean by "set the tracks select box"? is it that you want to create your own custom track selector?

**Reply by Jasonzzzzzzzzzzzzzz** [reply]
I try the tutorial and create a demo like this:
![screenshot20231020](https://github.com/GMOD/jbrowse-components/assets/49664139/12700f7e-d3d1-45e1-8f0e-06bf7b65103d)

In this demo, I have to click the button to open the track selector. I want the selector to default to the right, just like the first picture.

**Reply by cmdcolin** [reply, marked as answer]
currently the @jbrowse/react-linear-genome-view (which I think your screenshot shows) does not allow docking the track selector on the right hand drawer. the track selector is shown in a modal popup. I have considered that we could add ability for the embedded to show track selector as a drawer widget though. the @jbrowse/react-app actually allows using the drawer widget. see https://jbrowse.org/storybook/app/main/ for the @jbrowse/react-app demos

you can make a defaultSession with the track selector open, or programmatically use e.g. viewModel.session.views[0].activateTrackSelector()  or something like this

**Reply by Jasonzzzzzzzzzzzzzz** [reply]
Got it! It's really helpful. Thanks!
